### PR TITLE
[fix]: [CI-3643]: Add support for cross account assume role

### PR DESCRIPTION
**Pull request by shubham149**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31129)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/helper/ConnectorEnvVariablesHelper.java [hunk 1]
```diff
@@ -168,6 +170,15 @@
                 encodeBase64(String.valueOf(manualConfig.getSecretKeyRef().getDecryptedValue()))));
       }
     }
+
+    if (awsConnectorConfig.getCredential().getCrossAccountAccess() != null) {
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/helper/ConnectorEnvVariablesHelper.java:174]: What if it is via from delegate then credential can be null.

Comment (shubham149) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/helper/ConnectorEnvVariablesHelper.java]: added check for it

**Comment by shubham149** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 2 of 5 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31129)* status: 1. unresolved discussions.

**Pull request merged**